server/src-lib/Hasura/GraphQL/Schema/OrderBy.hs

```diff
@@ -64,28 +66,32 @@ orderByExp table selectPermissions = memoizeOn 'orderByExp table $ do
               otherTableParser <- lift $ orderByExp remoteTable perms
               pure $ do
                 otherTableOrderBy <- join <$> P.fieldOptional fieldName Nothing (P.nullable otherTableParser)
-                pure $ fmap (map $ fmap $ RQL.AOCObjectRelation relationshipInfo newPerms) otherTableOrderBy
+                pure $ fmap (map $ fmap $ IR.AOCObjectRelation relationshipInfo newPerms) otherTableOrderBy
             ArrRel -> do
               let aggregateFieldName = fieldName <> $$(G.litName "_aggregate")
               aggregationParser <- lift $ orderByAggregation remoteTable perms
               pure $ do
                 aggregationOrderBy <- join <$> P.fieldOptional aggregateFieldName Nothing (P.nullable aggregationParser)
-                pure $ fmap (map $ fmap $ RQL.AOCArrayAggregation relationshipInfo newPerms) aggregationOrderBy
+                pure $ fmap (map $ fmap $ IR.AOCArrayAggregation relationshipInfo newPerms) aggregationOrderBy
         FIComputedField _ -> empty
         FIRemoteRelationship _ -> empty
 
 
 
 -- local definitions
 
-type OrderInfo = (SQL.OrderType, SQL.NullsOrder)
+type OrderInfo = (PG.OrderType, PG.NullsOrder)
+
 
+-- FIXME!
+-- those parsers are directly using Postgres' SQL representation of
+-- order, rather than using a general intermediary representation
```

Comment:
Why is this bad?

Reply:
as we are building parsers that would parse a graphql query into an IR specific to Postgres? Are we planning on reusing the parser generation logic too across backends?

Reply:
Yeah, definitely, we will want to reuse as much as possible of the existing parsers. How much of that will be possible, I honestly don't know. We will probably want to use the same "trees that grow" logic, where parsers carry around some backend type information, so that they can delegate to backend-specific parsers when required... I can remove this FIXME if you prefer; we'll have to address this at some point anyway. ^^